src/mlpack/methods/cf/ncf_impl.hpp

```diff
@@ -186,12 +186,14 @@ void NCF::CreateGMF()
   userModel->Add<ann::Subview<> >(1, 0, size - 1);
   userModel->Add<ann::Embedding<> >(numUsers, embedSize);
   userModel->Add<ann::Subview<> >(size, 0, embedSize - 1, 0, size - 1);
+  //userModel->Add<ann::Linear<> >(embedSize*size, size);
```

Comment:
Isn't ``n_cols`` the number of samples and ``n_rows`` the number of features? In which case we would create a layer with the size of all samples?

Reply:
@zoq sorry for the late reply, i was travelling yesterday. Yes, dataset.n_cols mean the number of samples or user item pairs in the dataset. There aren't multiple features since it is just user and item data being used as input.